### ISSUE TITLE
Issues when using with realsense

### ISSUE THREAD

**Issue by SDADEEC** [leggedrobotics/raw_image_pipeline#9]
These issues are related to 'raw_image_pipeline_node.launch' launch files

1. If you set `input_type` to `color` and turn off all the image processing options (e.g., `debayer/enabled`, `flip/enabled`), the topics of debayered and color would still be published
And if you set `input_type` to `color` to other values, even if you turn on `debayer/enabled`, there is still nothing output

2. (I think this issue is related to the above one) If I set `flip/enabled` to `true` and turn off other options, the image would still be debayered. (Here the  `input_type` is set to `color`)

3. topics with postfix `slow` do not publish messages at a slow rate

4. Need feature to handle 16bit image (depth image)

**Comment by mmattamala** [comment]
Hi @SDADEEC , I added some changes to the branch [`fix/issue-9`](https://github.com/leggedrobotics/raw_image_pipeline/tree/fix/issue-9). Can you check if they work for you?

You can try the following settings for **depth images**:
```xml
  <!-- General options-->
  <arg name="camera_name"                      default="realsense_depth"/>
  <arg name="input_topic"                          default="/camera/aligned_depth_to_color/image_raw"/>
  <arg name="input_type"                           default="depth"/>

  <arg name="output_prefix"                        default="/camera_corrected"/>
  <arg name="output_frame"                         default="passthrough"/>
  
  <arg name="launch_prefix"                        default=""/>
  
  <arg name="output_encoding"                      default="passthrough"/>   <!-- Format of the published image BGR and RGB supported-->
  <arg name="skip_number_of_images_for_slow_topic" default="5"/>     <!-- To publish images at a slower rate -->
  <arg name="use_gpu"                              default="false"/>     <!-- To publish images at a slower rate -->
  <arg name="debug"                                default="false"/> <!-- to export debug data to /tmp -->

  <arg name="disable_compression_plugins"          default="true"/>     <!-- To avoid publication of theora or compressedDepth topics -->

  <!-- Modules -->
  <arg name="debayer/enabled"                      default="false"/>
  <arg name="flip/enabled"                               default="true"/>
  <arg name="white_balance/enabled"           default="false"/>
  <arg name="color_calibration/enabled"       default="false"/>
  <arg name="gamma_correction/enabled"   default="false"/>
  <arg name="vignetting_correction/enabled"        default="false"/>
  <arg name="color_enhancer/enabled"         default="false"/>
  <arg name="undistortion/enabled"              default="false"/>

  <!-- Camera -->
  <!-- Debayer  -->
  <arg name="debayer_encoding"                     default="auto"/>  <!-- auto, bayer_bggr8, bayer_gbrg8, bayer_grbg8, bayer_rggb8-->
   <!-- Flip  -->
  <arg name="flip/angle"                           default="270"/>  <!-- 90 (clockwise), 180, 270 (counter-clockwise) -->
```

An example of the output:
![image](https://github.com/leggedrobotics/raw_image_pipeline/assets/3932927/8cd73dea-24fc-4c50-892d-942785c58a7e)
![image](https://github.com/leggedrobotics/raw_image_pipeline/assets/3932927/63c91421-bef4-40fd-8b05-878c3f301f90)

**For color**, you only need to change these lines:
```xml
  <arg name="input_topic"                          default="/camera/color/image_raw"/>
  <arg name="input_type"                           default="color"/>
  ...
  <arg name="output_encoding"                default="RGB"/> 
```
The output should look like this
![image](https://github.com/leggedrobotics/raw_image_pipeline/assets/3932927/d1c53600-81fa-4ad9-8f6a-5d96a97e88fd)

**Comment by mmattamala** [comment]
> 5\. (ReadMe update) Need to add `glog_catkin` package to workspace to build

Regarding this part, this was in the README already: https://github.com/leggedrobotics/raw_image_pipeline#requirements-and-compilation
Perhaps I can make it more visible or move it upwards in the README

**Comment by mmattamala** [comment]
Closing this as it was merged in #11
